snapshot-labs/snapshot
# feat: prevent previously deleted ENS from being used again

### Summary

Closes: #4516

Prevent an ENS used by a deleted space to be used again for creating a new space.

### How to test

1. Connect with a wallet linked to a deleted space
2. Go to http://localhost:8080/#/setup?step=1
3. It should list all your wallet's available ENS
4. The ENS used by the deleted space should appear, but can not be used to create a new space 

With only valid domains, should list the ENS as before

![localhost_8080_ (2)](https://github.com/snapshot-labs/snapshot/assets/495709/7cd787ce-51b8-4cf7-9c7b-481443290114)

With at least one unavailable ENS domains:

![localhost_8080_ (1)](https://github.com/snapshot-labs/snapshot/assets/495709/2a30020d-703f-4501-aa97-012bf4c43497)

With multiple unavailable ENS domains:

![localhost_8080_ (3)](https://github.com/snapshot-labs/snapshot/assets/495709/a708c1ba-9ca2-4926-80b6-77559b330f6e)


### To-Do

- [x] Wait for https://github.com/snapshot-labs/snapshot-hub/pull/798 to be merged
- [x] Wait for the docs to be updated https://app.gitbook.com/o/-LFgTZvhAg63US8GVxGf/s/-MG4Ulnnabb2Xz3Lei9_/~/changes/460/~/overview
